File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/fragments/menus/LogfileMenuFragment.java
package de.dimedis.mobileentry.fragments.menus;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import de.dimedis.mobileentry.databinding.FragmentMenuLogfileBinding;
import de.dimedis.mobileentry.util.Logger;

public class LogfileMenuFragment extends AbsSettingsMenuFragment {

    FragmentMenuLogfileBinding binding;

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        public Builder() {
        }

        public LogfileMenuFragment build() {
            return new LogfileMenuFragment();
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        binding = FragmentMenuLogfileBinding.inflate(inflater, container, false);
        binding.okCancel.ok.setOnClickListener(view -> onClickOK());
        binding.okCancel.cancel.setOnClickListener(view -> onClickCancel());

        return binding.getRoot();
    }

    @Override
    protected void init() {
        super.init();
        binding.writeLocal.setChecked(mSettings.isWriteLogfile());
        binding.writeLocal.setOnCheckedChangeListener((buttonView, isChecked) -> {
            mSettings.setWriteLogfile(isChecked);
            if (isChecked)
                try {
                    Logger.clearLogFile();
                } catch (Exception ex) {
                    ex.printStackTrace();
                }
        });
    }

}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/fragments/menus/LogfileMenuFragment.kt
package de.dimedis.mobileentry.fragments.menus

import android.view.*
import de.dimedis.mobileentry.databinding.FragmentMenuLogfileBinding
import de.dimedis.mobileentry.util.*
import java.lang.Exception

import android.os.Bundle
import android.widget.CompoundButton

class LogfileMenuFragment : AbsSettingsMenuFragment() {
    var binding: FragmentMenuLogfileBinding? = null

    class Builder {
        fun build(): LogfileMenuFragment {
            return LogfileMenuFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        super.onCreateView(inflater, container, savedInstanceState)
        binding = FragmentMenuLogfileBinding.inflate(inflater, container, false)
        binding!!.okCancel.ok.setOnClickListener {
            onClickOK() }
        binding!!.okCancel.cancel.setOnClickListener {
            onClickCancel() }
        return binding!!.root
    }

    override fun init() {
        super.init()
        binding!!.writeLocal.isChecked = mSettings!!.writeLogfile
        binding!!.writeLocal.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            mSettings!!.writeLogfile = isChecked
            if (isChecked) try {
                Logger.clearLogFile()
            } catch (ex: Exception) {
                ex.printStackTrace()
            }
        }
    }

    companion object {
        fun builder(): Builder {
            return Builder()
        }
    }
}